code/modules/client/preferences.dm

```diff
@@ -1976,3 +1982,26 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			return
 		else
 			custom_names[name_id] = sanitized_name
+
+/// Handles adding and removing donator items from clients
+/datum/preferences/proc/handle_donator_items(client/user)
+	parent = user
+	var/req_donator = CONFIG_GET(flag/donator_items)
+	var/datum/loadout_category/DLC = GLOB.loadout_categories["Donator"] // stands for donator loadout category but the other def for DLC works too xD
+
+	if(IS_PATRON(user.ckey) || !req_donator)
+		for(var/key in DLC.gear)
+			var/datum/gear/donator/AG = GLOB.gear_datums[key]
+			if(!(AG.id in purchased_gear))
+				purchased_gear += AG.id
+				AG.purchase(user)
+		save_preferences()
+	else
+		if(purchased_gear.len || equipped_gear.len)
+			for(var/key in DLC.gear)
+				var/datum/gear/donator/RG = GLOB.gear_datums[key]
+				if(RG.id in equipped_gear)
+					equipped_gear -= RG.id
```

Comment:
This doesn't runtime if it isn't in the list, so you can remove both of the `if` checks and just do the `-=` without a care in the world (and avoid two string list lookups).